vms/avm/wallet_service.go

```diff
@@ -31,27 +31,26 @@ type WalletService struct {
 }
 
 func (w *WalletService) decided(txID ids.ID) {
-	if _, ok := w.pendingTxs.Get(txID); !ok {
+	if !w.pendingTxs.Delete(txID) {
 		return
 	}
 
 	w.vm.ctx.Log.Info("tx decided over wallet API",
 		zap.Stringer("txID", txID),
 	)
-	w.pendingTxs.Delete(txID)
-
 	for {
 		txID, tx, ok := w.pendingTxs.Oldest()
 		if !ok {
 			return
 		}
 
-		txBytes := tx.Bytes()
-		_, err := w.vm.IssueTx(txBytes)
+		err := w.vm.mempool.Add(tx)
```

Comment:
This is a behavior change. Previously we would validate the tx against the current state... now we just issue the tx into the mempool. Because this is only for transactions issued over the wallet API (which is deprecated) I think this minor change is fine.